### PR TITLE
[gym] Enable multi-GPU support with PPO when mirror_mat are defined

### LINKED ISSUE #773: [gym] Fix typing and docs RLlib utilities with ray >= 2.9
`algo.save()` doesn't return a checkpoint path anymore. Instead, we need to use `algo.save().checkpoint.path`. This directly impacts the `train` function in `gym_jiminy.rllib.utilities`

### PR DESCRIPTION
**Pull request by mwulfman**
Closes #773

### CODE REVIEW

**Comment by duburcqa** [comment]
You broke something else by trying to fix your bug. I suggest to keep the output of train as it is currently, but fix typing and doc, then adapt the other parts that assumes it returns the checkpoint path.

File: python/gym_jiminy/rllib/gym_jiminy/rllib/ppo.py [hunk 1]
```diff
@@ -155,6 +155,11 @@ def _compute_mirrored_value(value: torch.Tensor,
     """Compute mirrored value from observation space based on provided
     mirroring transformation.
     """
+    # Make sure value and mirror_mat are on the same device.
+    # This is needed for multi-GPU training.
+    if value.device != mirror_mat.device:
+        mirror_mat = mirror_mat.to(value.device)
```
Comment (duburcqa) [inline python/gym_jiminy/rllib/gym_jiminy/rllib/ppo.py]: no need to check the device. it is a no-op if unecessary.

File: python/gym_jiminy/rllib/gym_jiminy/rllib/utilities.py [hunk 2]
```diff
@@ -368,7 +369,7 @@ def train(algo: Algorithm,
           logdir: Optional[Logger] = None,
           checkpoint_period: int = 0,
           verbose: bool = True,
-          debug: bool = False) -> str:
+          debug: bool = False) -> _TrainingResult:
```
Comment (duburcqa) [inline python/gym_jiminy/rllib/gym_jiminy/rllib/utilities.py]: I already implemented a patch in my other PR. Maybe you could backport what I did here. I added some extra details.

File: python/gym_jiminy/rllib/gym_jiminy/rllib/ppo.py (hunk 1 above)
Comment (mwulfman) [reply python/gym_jiminy/rllib/gym_jiminy/rllib/ppo.py]: Fixed it.

File: python/gym_jiminy/rllib/gym_jiminy/rllib/utilities.py (hunk 2 above)
Comment (mwulfman) [reply python/gym_jiminy/rllib/gym_jiminy/rllib/utilities.py]: Ok, I reverted this change and renamed the PR.

**Pull request merged**